json_encoder_test.go

```diff
@@ -266,3 +268,19 @@ func TestJSONJSONEscaping(t *testing.T) {
 		assertJSON(t, output, enc)
 	}
 }
+
+func TestJSONOptions(t *testing.T) {
+	enc := newJSONEncoder(
```

Comment:
can you also verify that if you call `Clone` and use that `Clone` to log, you get the same output. (I want to make sure we test the clone's options are set correctly too)